src/Xamarin.Android.Linker/SetupStep.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Java.Interop.Tools.Cecil;
+using Mono.Linker;
+using Mono.Linker.Steps;
+using Mono.Tuner;
+using MonoDroid.Tuner;
+
+namespace Xamarin.Android.Linker
+{
+	class SetupStep : BaseStep
+	{
+		List<IStep> _steps;
+
+		List<IStep> Steps {
+			get {
+				if (_steps == null) {
+					var pipeline = typeof (LinkContext).GetProperty ("Pipeline").GetGetMethod ().Invoke (Context, null);
+					_steps = (List<IStep>) pipeline.GetType ().GetField ("_steps", BindingFlags.Instance | BindingFlags.NonPublic).GetValue (pipeline);
```

Comment:
Should we tell Marek Safar what you had to do here? Do we need some new public property here?

Reply:
I am going to add it to the "ILLinker Migration to .NET 5 APIs" document.

Reply:
This is an incorrect approach. All steps should set from msbuild target. We won't expose List<IStep> as that's an implementation detail which we are going to change

Reply:
Yes, I am aware of that. That is why I mentioned `Context.Pipeline` and `Pipeline` methods in the ILLinker-Migration-to-.NET-5-APIs document, to be considered for adding to the public API. This way we would not need reflection and access to the list itself.

Reply:
Discussed on Slack and we might not be able to have public Pipeline API. I would like to postpone the resolution for later. Opened https://github.com/xamarin/xamarin-android/issues/4709 for it. Will look into it in future PR.